library/src/scala/Tuple.scala

```diff
@@ -79,6 +83,17 @@ object Tuple {
     DynamicTuple.dynamicFromArray[Tuple](xs2)
   }
 
+  /** Convert an immutable array into a tuple of unknown arity and types */
+  def fromIArray[T](xs: IArray[T]): Tuple = {
+    val xs2: IArray[Object] = xs match {
+      case xs: IArray[Object] => xs
+      case xs =>
+        // TODO suport IArray.map
+        xs.asInstanceOf[Array[T]].map(_.asInstanceOf[Object]).asInstanceOf[IArray[Object]]
```

Comment:
Will a bound on `IArray` work, like `opaque type IArray[+T] <: Seq[T] = Array[_ <: T]`?

Reply:
Probably not. We need to do the same as array does and define a WrappedIArray for it.